packages/cli/test/unit/util/dev/parse-query-string.test.ts

```diff
@@ -0,0 +1,46 @@
+import {
+  parseQueryString,
+  formatQueryString,
+} from '../../../../src/util/dev/parse-query-string';
+
+describe('parseQueryString', () => {
+  it('should parse to Map and format back to original String', async () => {
+    const querystring = '?a&b=1&c=2&c=3&d=&space%20bar=4&html=%3Ch1%3E';
+    const parsed = parseQueryString(querystring);
+    const map = new Map<string, string[]>([
+      ['a', []],
+      ['b', ['1']],
+      ['c', ['2', '3']],
+      ['d', ['']],
+      ['space bar', ['4']],
+      ['html', ['<h1>']],
+    ]);
+    expect(parsed).toEqual(map);
```

Comment:
should these be strict/deep equals?

Reply:
This is jest which is deep equal already https://jestjs.io/docs/expect#toequalvalue